### PR TITLE
Migration to Play 2.4

### PR DESCRIPTION
**Pull request by bpervan**
Minor changes to migrate project to Play 2.4. Upgraded scala, sbt and sbt-plugin versions, switched Ebean from lib dependancy to sbt plugin

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 
-val appName         = "play2.3-rest-security"
+val appName         = "play2.4-rest-security"
 val appVersion      = "1.0"
 
-lazy val root = (project in file(".")).enablePlugins(PlayJava)
+lazy val root = (project in file(".")).enablePlugins(PlayJava, PlayEbean)
 
-scalaVersion := "2.11.1"
+scalaVersion := "2.11.6"
```
Comment (jamesward) [inline build.sbt]: I think `2.11.7` is the latest.

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version=0.13.5
+sbt.version=0.13.8
```
Comment (jamesward) [inline project/build.properties]: `0.13.9` is the latest.

File: project/plugins.sbt [hunk 3]
```diff
@@ -5,6 +5,8 @@ logLevel := Level.Warn
 resolvers += "Typesafe repository" at "http://repo.typesafe.com/typesafe/releases/"
 
 // Use the Play sbt plugin for Play projects
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.0")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.4.0")
```
Comment (jamesward) [inline project/plugins.sbt]: `2.4.6` is the latest.

File: project/plugins.sbt [hunk 4]
```diff
@@ -5,6 +5,8 @@ logLevel := Level.Warn
...
 
 addSbtPlugin("com.typesafe.sbt" % "sbt-coffeescript" % "1.0.0")
+
+addSbtPlugin("com.typesafe.sbt" % "sbt-play-ebean" % "1.0.0")
```
Comment (jamesward) [inline project/plugins.sbt]: I think `2.0.0` is the latest.

**Comment by jamesward** [comment]
I think we should do a full migration of this to the dependency injection stuff in Play 2.4.

**Comment by ddimaria** [comment]
@jamesward just wondering if you were planning on merging this PR for 2.4 and/or adding 2.4 DI code?

**Comment by jamesward** [comment]
I think this needs a full DI overhaul and migration to Play 2.5.  I'll work on that.

**Comment by jamesward** [comment]
I've bumped this to Play 2.5: https://github.com/jamesward/play-rest-security/commit/51e0e942b57b28932de46e0586cd1763cea659df

**Pull request closed without merging**